src/par_iter/chain.rs

```diff
@@ -199,14 +206,58 @@ impl<A, B> Producer for ChainProducer<A, B>
     }
 }
 
-impl<A, B> IntoIterator for ChainProducer<A, B>
-    where A: Producer,
-          B: Producer<Item = A::Item>
+/// ////////////////////////////////////////////////////////////////////////
+/// Wrapper for Chain to implement ExactSizeIterator
+
+pub struct Chain<A, B> {
+    chain: iter::Chain<A, B>,
+    len: usize,
+}
+
+impl<A, B> Chain<A, B> {
+    fn new(a: A, b: B) -> Chain<A, B>
+        where A: ExactSizeIterator,
+              B: ExactSizeIterator<Item=A::Item>
+    {
+        let len = a.len() + b.len();
```

Comment:
@cuviper I'm not sure I understand: are you suggesting that this wrapper's `size_hint` just use use the result from the underlying `std::iter::Chain::size_hint`? The issue I see with that is I don't know whether that underlying value would always be accurate (since the only reason for this wrapper to exist is that the original `Chain` doesn't implement `ExactSizeIterator`).